Wordle.Domain/Guess.cs

```diff
@@ -2,16 +2,21 @@ namespace Wordle.Domain;
 
 public class Guess
 {
-    public WordScore[] Guesses = new WordScore[6];
+    public Dictionary<int, WordScore> Guesses = new Dictionary<int, WordScore>(6);
     public int GuessCount = 0;
-    public void UpdateGuesses(List<LetterScore> letterScores)
+    public bool TryUpdateGuesses(string answer, string guess)
```

Comment:
I like what you did here.  

Thought/Suggestion: can you think of a more appropriate name for this function?  Since it returns a boolean, `Try...` doesn't make me think boolean.  I believe the general naming convention for boolean returns starts with `Is` or "answers" a yes-no question, though I am open to a debate on that! :) Maybe `IsGuessUpdated` or something along those lines?  What do you think?

Reply:
This is a great call out, this is the [official Microsoft naming recommendation](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members?redirectedfrom=MSDN) for boolean properties:

> DO name Boolean properties with an affirmative phrase (CanSeek instead of CantSeek). Optionally, you can also prefix Boolean properties with "Is", "Can", or "Has", but only where it adds value.
I like the idea of using `CanUpdateGuesses` or `IsGuessUpdated` equally, but since I use Is for boolean properties in other places I think I'll keep it consistent!